src/python/pants/engine/fs_test.py

```diff
@@ -957,24 +930,79 @@ async def workspace_goal_rule(
     assert Path(rule_runner.build_root, "a.txt").read_text() == "hello"
 
 
-def test_write_digest() -> None:
-    rule_runner = RuleRunner()
+# -----------------------------------------------------------------------------------------------
+# Invalidation of the FS
+# -----------------------------------------------------------------------------------------------
```

Comment:
These tests were greatly simplified. They are still testing the same thing, but with much less boilerplate and fewer helper functions.